File: golang/2078.two-furthest-houses-with-different-colors.go
func maxDistance(colors []int) int {
	n := len(colors)
	left, right := 0, n-1

	for left < n && colors[left] == colors[n-1] {
		left++
	}

	for right >= 0 && colors[right] == colors[0] {
		right--
	}

	return max(right, n-left-1)
}

File: python/2078.two-furthest-houses-with-different-colors.py
class Solution:
    def maxDistance(self, colors: list[int]) -> int:

        n = len(colors)

        left = 0
        right = n - 1

        while left < n and colors[left] == colors[n - 1]:
            left += 1

        while right >= 0 and colors[right] == colors[0]:
            right -= 1

        return max(right, n - left - 1)
